gprofiler/main.py

```diff
@@ -551,7 +563,10 @@ def parse_cmd_args():
         parser.error("--perf-dwarf-stack-size maximum size is 65528")
 
     if args.perf_mode in ("dwarf", "smart") and args.frequency > 100:
-        parser.error("--profiling-frequency|-f can't be larger than 100 when using --perf-mode smart|dwarf")
+        parser.error("--profiling-frequency|-f can't be larger than 100 when using --perf-mode 'smart' or 'dwarf'")
```

Comment:
Hmm first time I notice this.
I don't think its a good idea to deny users the option to sample at a higher frequency. We could instead warn about disk space implications.

Reply:
It's not about disk space but about CPU usage. Anyway, that limit is indeed less relevant, we don't want to deny a conscious user from doing whatever they want. Will remove in another PR